node/assemble_test.go

```diff
@@ -161,3 +162,74 @@ func BenchmarkAssembleBlock(b *testing.B) {
 		require.True(b, found)
 	}
 }
+
+type callbackLogger struct {
+	logging.Logger
+	WarnfCallback func(string, ...interface{})
+}
+
+func (cl callbackLogger) Warnf(s string, args ...interface{}) {
+	cl.WarnfCallback(s, args...)
+}
+
+func TestAssembleBlockTransactionPoolBehind(t *testing.T) {
+	const numUsers = 100
+	expectingLog := false
+	baseLog := logging.TestingLog(t)
+	baseLog.SetLevel(logging.Info)
+	log := &callbackLogger{
+		Logger: baseLog,
+		WarnfCallback: func(s string, args ...interface{}) {
+			require.True(t, expectingLog)
```

Comment:
I think it would be a good idea to reset the expectingLog after that statement, right ?
i.e. to ensure we don't get 5 calls with that content.